internal/service/mongodb/mongodb_users.go

```diff
@@ -453,11 +390,133 @@ func convertToCloudMongodbUser(values basetypes.ListValue) []*vmongodb.DeleteClo
 	return result
 }
 
-func containsInUsergList(userName string, users []*vmongodb.CloudMongoDbUser) bool {
-	for _, v := range users {
-		if *v.UserName == userName {
-			return true
+func convertToSingleAddOrChangeParameter(v MongodbUser) *vmongodb.AddOrChangeCloudMongoDbUserParameter {
+	param := &vmongodb.AddOrChangeCloudMongoDbUserParameter{
+		UserName:     v.UserName.ValueStringPointer(),
+		Password:     v.Password.ValueStringPointer(),
+		DatabaseName: v.DatabaseName.ValueStringPointer(),
+		Authority:    v.Authority.ValueStringPointer(),
+	}
+
+	return param
+}
+
+func convertToSingleDeleteParameter(v MongodbUser) *vmongodb.DeleteCloudMongoDbUserParameter {
+	param := &vmongodb.DeleteCloudMongoDbUserParameter{
+		UserName:     v.UserName.ValueStringPointer(),
+		DatabaseName: v.DatabaseName.ValueStringPointer(),
+	}
+
+	return param
+}
+
+func addOrChangeUserList(ctx context.Context, config *conn.ProviderConfig, id *string, planUserList []MongodbUser, stateUserList []MongodbUser) error {
+	changeParameters := make([]*vmongodb.AddOrChangeCloudMongoDbUserParameter, 0)
+	addParameters := make([]*vmongodb.AddOrChangeCloudMongoDbUserParameter, 0)
+
+	for _, pv := range planUserList {
+		found := false
+		for _, sv := range stateUserList {
+			if pv.UserName.Equal(sv.UserName) && pv.DatabaseName.Equal(sv.DatabaseName) {
+				if !pv.Password.Equal(sv.Password) || !pv.Authority.Equal(sv.Authority) {
+					changeParameters = append(changeParameters, convertToSingleAddOrChangeParameter(pv))
+				}
+				found = true
+				break
+			}
+		}
+
+		if !found {
+			addParameters = append(addParameters, convertToSingleAddOrChangeParameter(pv))
+		}
+	}
+
+	if len(changeParameters) > 0 {
+		reqParams := &vmongodb.ChangeCloudMongoDbUserListRequest{
+			RegionCode:             &config.RegionCode,
+			CloudMongoDbInstanceNo: id,
+			CloudMongoDbUserList:   changeParameters,
+		}
+		tflog.Info(ctx, "ChangeCloudMongoDbUserList reqParams="+common.MarshalUncheckedString(reqParams))
```

Comment:
I'm guessing that the password info set by the user might be included in the logs. If so, we'll need to get rid of it not exposed.